src/main/java/com/massivecraft/factions/listeners/FactionsPlayerListener.java

```diff
@@ -102,6 +102,11 @@ public void run() {
             P.p.log(Level.INFO, "Found %s on admin Bypass without permission on login. Disabled it for them.", player.getName());
         }
 
+        // If they last logged off with faction fly enabled, we can re-enable it so they don't fall and die
+        if (P.p.getConfig().getBoolean("f-fly.enable-flight-on-login") && me.isFlying() || me.isAutoFlying()) {
+            me.setFlying(true);
```

Comment:
Shouldn't we check if they are not on the ground?

Reply:
That's sort of the point of the `me.isFlying()` is it not? It'll check whether or not the player is flying and if so set the player to actually being flying.

Reply:
That checks their FPlayer if they are set to currently flying (were flying on logout). It doesn't check their Bukkit player to see if they are actually flying or not.

Reply:
Added a check in https://github.com/minksuki/Factions/commit/cf70de1170cc7d2e7f601cf57fb8776a11e04b2f. Thanks for the suggestion.